include/FileBrowserSearcher.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * FileBrowserSearcher.h - Batch processor for searching the filesystem
+ *
+ * Copyright (c) 2023 saker <sakertooth@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_FILE_BROWSER_SEARCHER_H
+#define LMMS_FILE_BROWSER_SEARCHER_H
+
+#include <QHash>
+#include <QString>
+#include <QStringList>
+#include <optional>
+#include <queue>
+
+#ifdef __MINGW32__
+#include <mingw.condition_variable.h>
+#include <mingw.mutex.h>
+#include <mingw.thread.h>
+#else
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+#endif
+
+namespace lmms::gui {
+class FileBrowserSearcher
+{
+public:
+	static constexpr int s_batchSize = 16;
+	static constexpr int s_millisecondsPerBatch = 500;
```

Comment:
```suggestion
	static constexpr int BatchSize = 16;
	static constexpr int MillisecondsPerBatch = 500;
```
I would use the naming convention for constants here, since these are public members.

`FileBrowserSearcher::s_millisecondsPerBatch` looks like something you're not supposed to do, like you're accessing a private member that was accidentally made public. But `FileBrowserSearcher::MillisecondsPerBatch` looks fine, and it's how we do it elsewhere in the codebase.